.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,70 @@
+# This is a basic workflow to help you get started with Actions
+
+name: benchmark-check
+
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on push or pull request events but only for the master branch
+  pull_request:
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  build:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      # This is too complicated to get working for now.
```

Comment:
Is [the way Cirrus does this](https://cirrus-ci.org/guide/tips-and-tricks/#custom-clone-command) any help/inspiration?

Reply:
Thanks for the link, I'll play with a version of that and see if I can get it to work.  asv calls a variety of git commands when it is writing out the benchmarks and not all of them have been happy with a shallow clone.